### PR TITLE
Allow all door access in Suspicion mode

### PR DESCRIPTION
**Pull request by Visne**

### CODE REVIEW

File: Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs [hunk 1]
```diff
@@ -155,9 +161,27 @@ public bool CanOpen(IEntity user)
...
+        }
+
         public void TryOpen(IEntity user)
         {
-            if (!CanOpen(user))
+            var isSuspicion = _gameTicker.HasGameRule(typeof(RuleSuspicion));
+            var isAirlockExternal = HasAccess("External");
+
+            // Only open if CanOpen and not Suspicion OR if not external airlock in suspicion.
+            if ((!CanOpen(user) && !isSuspicion) || (isSuspicion && isAirlockExternal))
```
Comment (Zumorica) [inline Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs:184]: This is a terrible way to solve this problem.

**Review by Zumorica — CHANGES_REQUESTED** [review]

**Comment by Visne** [comment]
Important note: removing the gamerule Suspicion will set the door access to ID. If another gamerule had set it to something else, it will still change it.

**Comment by Visne** [comment]
Last comment related to #1899

File: Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs [hunk 2]
```diff
@@ -142,20 +143,46 @@ public bool CanOpen(IEntity user)
...
-            if (user.TryGetComponent(out HandsComponent? hands) && hands.Count == 0)
+            var access = doorSystem.AccessType switch
             {
-                EntitySystem.Get<AudioSystem>().PlayFromEntity("/Audio/Effects/bang.ogg", Owner,
-                    AudioParams.Default.WithVolume(-2));
+                DoorSystem.AccessTypes.AllowAll => true,
+                DoorSystem.AccessTypes.AllowAllIdExternal => isAirlockExternal ? CanOpen(user) : true,
+                DoorSystem.AccessTypes.AllowAllNoExternal => !isAirlockExternal,
+                _ => CanOpen(user)
+            };
```
Comment (DrSmugleaf) [inline Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs:171]: The access type check should probably be moved into `CanOpen(IEntity)`, unless you specifically want that method to not affect overriding classes.

File: Content.Server/GameObjects/EntitySystems/DoorSystem.cs [hunk 3]
```diff
@@ -1,10 +1,45 @@
...
 
 namespace Content.Server.GameObjects.EntitySystems
 {
+    [UsedImplicitly]
     class DoorSystem : EntitySystem
     {
+        /// <summary>
+        /// Allows all players to open airlocks, despite not having access. External airlocks, so airlocks that are
+        /// open to space, are not included.
+        /// </summary>
```
Comment (DrSmugleaf) [inline Content.Server/GameObjects/EntitySystems/DoorSystem.cs:13]: This summary seems like an incorrect copy-paste from AccessTypes.AllowAll.

```suggestion
        /// <summary>
        ///     Determines the base access behavior of all doors on the station.
        /// </summary>
```

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

File: Content.Server/GameObjects/EntitySystems/DoorSystem.cs (hunk 3 above)
Comment (Visne) [reply Content.Server/GameObjects/EntitySystems/DoorSystem.cs:13]: Oops, that was changed from a boolean into an enum but I forgot to change the summary.

File: Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs (hunk 2 above)
Comment (Visne) [reply Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs:171]: Yes, that would make more sense. I'll move it.

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**